cmd/tdex/depositmarket.go

```diff
@@ -31,29 +60,553 @@ var depositmarket = cli.Command{
+		txID = resp
+		break
+	}
+
+	outpoints := make([]*pboperator.TxOutpoint, 0, len(outputs))
+	for i := 0; i < len(outputs); i++ {
+		outpoints = append(outpoints, &pboperator.TxOutpoint{
+			Hash:  txID,
+			Index: int32(i),
+		})
+	}
+
+	//wait one min so that tx get confirmed
+	time.Sleep(5 * time.Second)
+
+	_, err = client.ClaimMarketDeposit(
+		context.Background(), &pboperator.ClaimMarketDepositRequest{
+			Market: &types.Market{
+				BaseAsset:  assetValuePair.BaseAsset,
+				QuoteAsset: assetValuePair.QuoteAsset,
+			},
+			Outpoints: outpoints,
+		},
+	)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
+
+func fetchMarketAddresses(
+	outsLen int,
+	client pboperator.OperatorClient,
+	baseAssetOpt string,
+	quoteAssetOpt string,
+) ([]string, error) {
+	depositMarket, err := client.DepositMarket(
+		context.Background(), &pboperator.DepositMarketRequest{
+			Market: &pbtypes.Market{
+				BaseAsset:  baseAssetOpt,
+				QuoteAsset: quoteAssetOpt,
+			},
+			NumOfAddresses: int64(outsLen),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return depositMarket.GetAddresses(), nil
+}
+
+func createOutputs(
+	baseFragments, quoteFragments []uint64,
+	feeAmount uint64,
+	addresses []string,
+	assetValuePair AssetValuePair,
+	baseAssetKey string,
+) []TxOut {
+	outsLen := len(baseFragments) + len(quoteFragments)
+	outputs := make([]TxOut, 0, outsLen)
+
+	index := 0
+	for i, v := range baseFragments {
+		value := int64(v)
+		//deduct fee from last(largest) fragment
+		if i == len(baseFragments)-1 {
+			value = int64(v) - int64(feeAmount)
+		}
+		outputs = append(outputs, TxOut{
+			Asset:   baseAssetKey,
+			Value:   value,
+			Address: addresses[index],
+		})
+		index++
+	}
+	for _, v := range quoteFragments {
+		outputs = append(outputs, TxOut{
+			Asset:   assetValuePair.QuoteAsset,
+			Value:   int64(v),
+			Address: addresses[index],
+		})
+		index++
+	}
+
+	return outputs
+}
+
+func fragmentUnspents(pair AssetValuePair) ([]uint64, []uint64) {
+
+	baseAssetFragments := make([]uint64, 0)
+	quoteAssetFragments := make([]uint64, 0)
+
+	baseSum := uint64(0)
+	quoteSum := uint64(0)
+	for numOfUtxo, percentage := range fragmentationMap {
+		for ; numOfUtxo > 0; numOfUtxo-- {
+			baseAssetPart := percent(int(pair.BaseValue), percentage)
+			baseSum += uint64(baseAssetPart)
+			baseAssetFragments = append(baseAssetFragments, uint64(baseAssetPart))
+
+			quoteAssetPart := percent(int(pair.QuoteValue), percentage)
+			quoteSum += uint64(quoteAssetPart)
+			quoteAssetFragments = append(quoteAssetFragments, uint64(quoteAssetPart))
+		}
+	}
+
+	sort.Slice(baseAssetFragments, func(i, j int) bool {
+		return baseAssetFragments[i] < baseAssetFragments[j]
+	})
+
+	sort.Slice(quoteAssetFragments, func(i, j int) bool {
+		return quoteAssetFragments[i] < quoteAssetFragments[j]
+	})
+
+	//if there is rest, created when calculating percentage,
+	//add it to last fragment
+	if baseSum != pair.BaseValue {
+		baseRest := pair.BaseValue - baseSum
```

Comment:
could it happen that baseSum is actually > pair.BaseValue? it shouldn't ever happen I guess, but I don't see any code that enforces this.

Reply:
It cant, cause baseSum is sum of parts(percentages) of pair.BaseValue